packages/wonder-blocks-badge/src/components/badge.tsx

```diff
@@ -1,5 +1,202 @@
+import {addStyle, AriaProps} from "@khanacademy/wonder-blocks-core";
+import {styles as typographyStyles} from "@khanacademy/wonder-blocks-typography";
+import {border, semanticColor, sizing} from "@khanacademy/wonder-blocks-tokens";
+import {StyleSheet} from "aphrodite";
 import * as React from "react";
 
-export const Badge = () => {
-    return <div>Badge component</div>;
+type Props = AriaProps & {
+    /**
+     * The kind of badge to display. Defaults to `info`.
+     */
+    kind?: "info" | "success" | "warning" | "critical";
+    /**
+     * The label to display in the badge.
+     */
+    label?: string;
+    /**
+     * The icon to display in the badge. It can be a PhosphorIcon, a custom svg,
+     * or `img` element.
+     * Considerations:
+     * - If the icon conveys meaning, set the alt text on the icon being used
+     * - If the icon is an `img` element, it may need width: 100% and height: 100%
+     *   to render properly in the badge.
+     */
+    icon?: React.ReactElement;
 };
+
+const StyledDiv = addStyle("div");
+const StyledSpan = addStyle("span");
+
+function getKindStyle(kind: Props["kind"]) {
+    switch (kind) {
+        case "info":
+            return styles.info;
+        case "success":
+            return styles.success;
+        case "warning":
+            return styles.warning;
+        case "critical":
+            return styles.critical;
+        default:
+            return {};
+    }
+}
+
+function getIconKindStyle(kind: Props["kind"]) {
+    switch (kind) {
+        case "info":
+            return styles.infoIcon;
+        case "success":
+            return styles.successIcon;
+        case "warning":
+            return styles.warningIcon;
+        case "critical":
+            return styles.criticalIcon;
+        default:
+            return {};
+    }
+}
+
+/**
+ * Badges are visual indicators used to display concise information, such as
+ * a status, label, or count.
+ */
+export const Badge = (props: Props) => {
+    const {kind = "info", icon, label, ...otherProps} = props;
+    if (!label && !icon) {
+        return <React.Fragment />;
+    }
+    return (
+        <StyledDiv
+            style={[
+                typographyStyles.LabelXSmall,
+                styles.badge,
+                getKindStyle(kind),
+                icon && !label ? styles.iconOnly : {},
+            ]}
+            {...otherProps}
+        >
+            {icon && (
+                <StyledSpan style={[styles.icon, getIconKindStyle(kind)]}>
+                    {icon}
+                </StyledSpan>
+            )}
+            {label}
+        </StyledDiv>
+    );
+};
+
+const badgeTokens = {
```

Comment:
Setting up a structure for tokens related to badges. Let me know if anyone has any feedback on this - I was trying to follow the [naming guide](https://khanacademy.atlassian.net/wiki/x/FwB47)!

This doesn't reflect the TB styling yet. We can make updates to the tokens used (or if we need to have component themes) once the TB theme is in place!